README.md

```diff
@@ -1,63 +1,145 @@
 
-#### [Carthage](https://github.com/carthage/carthage)
+## Usage
 
-Add the following line to your Cartfile:
+```swift
+import MapboxDirections
+import MapboxNavigationUI
+```
 
-```ruby
-github "mapbox/mapbox-navigation-ios"
+```swift
+let origin = Waypoint(coordinate: CLLocationCoordinate2D(latitude: 38.9131752, longitude: -77.0324047), name: "Mapbox")
+let destination = Waypoint(coordinate: CLLocationCoordinate2D(latitude: 38.8977, longitude: -77.0365), name: "White House")
+        
+let options = RouteOptions(waypoints: [origin, destination], profileIdentifier: .automobileAvoidsTraffic)
+options.includesSteps = true
+
+Directions.shared.calculate(options) { (waypoints, routes, error) in
+    guard let route = routes?.first else { return }
+    
+    let viewController = NavigationUI.routeViewController(for: route)
+    self.present(viewController, animated: true, completion: nil)
+}
+```
+
+See [this guide](https://github.com/mapbox/mapbox-navigation-ios/blob/master/Docs/Storyboards.md) for usage with storyboards.
+
+### UI overrides and listeners
+
+#### Colors
+
+You can override the default colors in the UI.
+
+```swift
+// Used for guidance arrow, highlighted text and progress bars.
+NavigationUI.shared.tintColor = .red
+
+// Used for guidance arrow
+NavigationUI.shared.tintStrokeColor = .blue
+
+// Used for titles and prioritized information
+NavigationUI.shared.primaryTextColor = .orange
+
+// Used for subtitles, distances and accessory labels
+NavigationUI.shared.secondaryTextColor = .pink
+
+// Used for separators in table views
+NavigationUI.shared.lineColor = .yellow
 ```
 
-### Gist of how this works
+#### RouteViewController Delegate Methods
+
+* `routeControllerDidCancelNavigation`: Fired when the user taps `Cancel`. You are responsible for dismissing the UI
+
+## Examples
+
+We provide examples in Swift and Objective-C. Run `carthage update --platform ios` from the root folder and open MapboxNavigation.xcodeproj to try it out.
+
+### Running the example app
+
+1. If running in the simulator, you can simulate the user location by selecting `Debug` -> `Location` -> `City Bicycle Ride`
+1. Long press any where on a map. This is where you will be routed to.
+1. Press `Start Navigation` to begin
+1. Press `Cancel` to end
+
+## Building your own custom navigation UI
 
-`RouteController` is given a route. Internally, MapboxNavigation.swift is comparing the route to the users location and looking at 3 principle pieces:
+Mapbox Navigation gives you all the components you need, should you want to build your own custom turn-by-turn navigation UI:
+
+* [Mapbox Maps SDK](https://www.mapbox.com/ios-sdk/)
+* [Mapbox Studio](https://www.mapbox.com/studio/)
+  * Design custom maps with live traffic overlays
+* [MapboxDirections](https://github.com/mapbox/MapboxDirections.swift)
+  * Automotive, cycling, and walking directions
+  * Traffic-influenced driving directions
+* Mapbox Navigation Core Components
+  * Route Controller
```

Comment:
Nit: sentence case is fine for these two lines. We aren’t treating Mapbox Navigation Core Components as a product in its own right, just another of our bazillions of public helper libraries.